openlibrary/core/bookshelves.py

```diff
@@ -202,6 +202,97 @@ def count_total_books_logged_by_user_per_shelf(
         result = oldb.query(query, vars=data)
         return {i['bookshelf_id']: i['count'] for i in result} if result else {}
 
+    # Iterates through a list of solr docs, and for all items with a 'logged edition'
+    # it will remove an item with the matching edition key from the list, and add it to
+    # doc["editions"]["docs"]
+    def link_editions_to_works(solr_docs):
+        linked_docs: list[web.storage] = []
+        docs_dict = {}
+        # adds works to linked_docs, recording their edition key and index in docs_dict if present.
+        for doc in solr_docs:
+            full_key = doc.get("key") or "/dummy/value"
```

Comment:
This should always have a key, so to more clearly communicate that let's drop the default. If in some very strange case it is missing the key, we likely want it to error!
```suggestion
            full_key = doc["key"]
```